src/Microsoft.DocAsCode.Build.Engine/SingleDocumentBuilder.cs

```diff
@@ -932,6 +936,16 @@ private IMarkdownService CreateMarkdownService(DocumentBuildParameters parameter
                 });
         }
 
+        private static string ComputeConfigHash(DocumentBuildParameters parameter)
+        {
+            using (var ms = new MemoryStream())
+            using (var writer = new StreamWriter(ms))
+            {
+                writer.Write(JsonUtility.Serialize(parameter));
+                return Convert.ToBase64String(MD5.Create().ComputeHash(ms.ToArray()));
```

Comment:
>Convert.ToBase64String [](start = 23, length = 22)

extract in some utility to calculate the hash by md5? pass string to be computed as parameter?